rdio-call

```diff
@@ -21,99 +21,108 @@ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-
-# create the Rdio client object
-rdio = Rdio(config['consumer_key'], config['consumer_secret'], config['auth_state'], **extra_constructor_arguments)
-
-# authenticate, if requested
-if options.authenticate:
-  import webbrowser
-  webbrowser.open(rdio.begin_authentication('oob'))
-  verifier = raw_input('Enter the PIN from the Rdio site: ').strip()
-  rdio.complete_authentication(verifier)
-
-# save state
-with file(config_path, 'w') as f:
-  json.dump(config, f, indent=True)
-  f.write('\n')
-
-# process the response
-response, content = rdio.call_raw(method, **args)
-if response['status'] == '200':
-  if options.indent:
-    json.dump(json.loads(content), sys.stdout, indent=True)
-    sys.stdout.write('\n')
+
+def main(options, args):
+  """Make a call to Rdio."""
+  # load the persistent state
+  config_path = os.path.expanduser('~/.rdio-tool.json')
+  if os.path.exists(config_path):
+    config = json.load(file(config_path))
   else:
-    print content
-else:
-  if options.verbose:
-    if response.has_key('www-authenticate'):
-      print 'Authentication required, pass --authenticate'
+    config = {'auth_state': {}}
+
+  # set up the keys
+  if options.client_key is not None:
+    config['client_key'] = options.client_key
+  if options.client_secret is not None:
+    config['client_secret'] = options.client_secret
+
+  urls = {
+    'api_endpoint': options.api_endpoint,
+    'device_code_url': options.device_code_url,
+    'token_url': options.token_url
+  }
+
+  if 'client_key' not in config or 'client_secret' not in config:
+    sys.stderr.write('Both the client key and client secret must be specified')
+    sys.exit(1)
+
+  # forget auth state, if we're logging in or logging out
+  if options.forget_auth or options.authenticate:
+    config['auth_state'] = {}
+
+  # create the Rdio client object
+  rdio = Rdio(config['client_key'], config['client_secret'], config['auth_state'], urls)
+
+  # authenticate, if requested
+  if options.authenticate:
+    url, device_code = rdio.begin_authentication()
+    print 'Please enter device code: %s on %s' % (device_code, url)
+    webbrowser.open(url)
+    rdio.complete_authentication()
+
+  # save state
+  with file(config_path, 'w') as f:
+    json.dump(config, f, indent=True)
+    f.write('\n')
+
+  # process the response
+  response, content = rdio.call_raw(method, **args)
+  if response['status'] == '200':
+    if options.indent:
+      json.dump(json.loads(content), sys.stdout, indent=True)
+      sys.stdout.write('\n')
     else:
       print content
-  sys.exit(int(response['status']))
+  else:
+    if options.verbose:
+      if 'www-authenticate' in response:
+        print 'Authentication required, pass --authenticate'
+      else:
+        print content
+    sys.exit(int(response['status']))
+
+if __name__ == "__main__":
+  parser = OptionParser(usage='%prog [options] method arg1=value1 arg2=value2...', version='%prog 0.1')
+  parser.add_option('--client-key', dest='client_key',
```

Comment:
Please call it _client ID_ instead of _client key_ to be consistent with the API docs